packages/mdc-auto-init/index.js

```diff
@@ -19,6 +19,29 @@ const registry = Object.create(null);
 const CONSOLE_WARN = console.warn.bind(console);
 
 /**
+ * Fires a cross-browser-compatible custom event from HTML element or HTML document,
+ * with the given data.
+ * @param {HTMLElement, HTMLDocument} element
+ * @param {string} evtType
+ * @param {!Object} evtData
+ * @param {boolean=} shouldBubble
+ */
+export function emit(element, evtType, evtData, shouldBubble = false) {
```

Comment:
IMHO, I might not consider making emit a **private** helper function of `mdc-auto-init` instead of public. So far, I cannot see a strong argument why this method need to be public.

If testing is your biggest concern, how about cover both branch using the following methods?
The logic behind this is, we don't need explicitly export `emit` to test it but we mock the run-time environment (as you did in testing `emit` directly) and see if `mdcAutoInit` behave uniformly in both case.
```
test('#emit event from MDCComponent on init complete', () => {
  const root = setupTest();
  const handler = td.func('init event handler');
  let evt = null;

  td.when(handler(td.matchers.isA(Object))).thenDo((evt_) => {
    evt = evt_;
  });

  const type = 'MDCAutoInit:End';

  document.addEventListener(type, handler);
  mdcAutoInit(root);

  assert.isOk(evt !== null);
  assert.equal(evt.type, type);
});

test('#emit event from MDCComponent on init complete - custom events not supported', () => {
  const root = setupTest();
  const handler = td.func('init event handler');
  let evt = null;

  td.when(handler(td.matchers.isA(Object))).thenDo((evt_) => {
    evt = evt_;
  });

  const type = 'MDCAutoInit:End';

  document.addEventListener(type, handler);
  const {CustomEvent} = window;
  window.CustomEvent = undefined;
  try {
    mdcAutoInit(root);
  } finally {
    window.CustomEvent = CustomEvent;
  }

  assert.isOk(evt !== null);
  assert.equal(evt.type, type);
});
```

Let me know if this makes sense or your consideration for making `emit` public. Thanks!

Reply:
Sure. Your proposal is better. I exported `emit` just because of the test.
Do you think I should remove the `element` param from the emit function and just use `document`? I don't think `element` param makes sense anymore because tests always attach the listener to document.

Reply:
@yshterev Sorry for the tardiness. Yes. I think that is totally a valid suggestion, right now we are only emitting event at document level, which means the `element` is not really being used. Feel free to proceed with this idea and let me know when it is ready to review.